src/MultiRegionClient.php

```diff
@@ -45,6 +51,25 @@ public static function getArguments()
                     };
                 },
             ],
+            'partition' => [
+                'type'    => 'config',
+                'valid'   => ['string', Partition::class],
+                'doc'     => 'AWS partition to connect to. Valid partitions include "aws," "aws-cn," and "aws-us-gov."',
+                'default' => function (array $args) {
+                    $region = isset($args['region']) ? $args['region'] : '';
+                    return PartitionEndpointProvider::defaultProvider()
+                        ->getPartition($region);
+                },
+                'fn'      => function ($value, array &$args) {
+                    if (is_string($value)) {
+                        $value = PartitionEndpointProvider::defaultProvider()
+                            ->getPartition($value);
```

Comment:
Isn't this trying to get a partition by partition name, but the method only works with region names?